core/history.go

```diff
@@ -11,27 +11,27 @@ import (
 
 var ErrCheckHeadState = errors.New("check head state")
 
-type History struct {
+type history struct {
 	txn db.Transaction
 }
 
-func NewHistory(txn db.Transaction) *History {
-	return &History{txn: txn}
+func NewHistory(txn db.Transaction) *history {
```

Comment:
we should either make this private or remove it all together

Reply:
If we make that private, how to handle the `history_test.go` then? @omerfirmak

Reply:
cc @IronGauntlets

Reply:
> we should either make this private or remove it all together

 @Atoo35 Please remove it

>If we make that private, how to handle the history_test.go then?

Please rename the test file to `history_pkg_test.go` (this is not a go convention, just over convention) and change the package name in `history_test.go` from `core_test` to `core`.

Reply:
How is removing helping? the history test package will error out entirely and so will the state.go. Am I missing something here?

Reply:
There is no need to have a `NewHistory()` as it is a trivial func. Anywhere `NewHistory()` is used it can be replaced with `&history{txn: txn}`. You should only create `New` func when you want to assert all the arguments required to create an instance of the struct, or you do some extra work or if the struct is private and the only way to instantiate is through a public `New` func.